src/components/views/rooms/MemberList.js

```diff
@@ -196,6 +196,9 @@ export default class MemberList extends React.Component {
             event.getType() === "m.room.third_party_invite") {
             this._updateList();
         }
+        if (event.getContent().invite !== event.getPrevContent().invite) {
+            this.forceUpdate();
```

Comment:
`forceUpdate` is a smell, `canJoin` should be moved into state.
Additionally keep in mind this isn't the only thing that can change whether or not you can invite someone, if your power level changes that will affect it too

Reply:
> `canJoin` should be moved into state.

Oh yes, that's a better idea. 
> if your power level changes that will affect it too

Forgot about that. Will modify.